components/size_aware_image.jsx

```diff
@@ -55,28 +54,16 @@ export default class SizeAwareImage extends React.PureComponent {
 
     componentDidMount() {
         this.mounted = true;
-        this.loadImage();
-    }
-
-    componentDidUpdate(prevProps) {
-        if (this.props.src !== prevProps.src) {
-            this.loadImage();
-        }
     }
 
     componentWillUnmount() {
         this.mounted = false;
     }
 
-    loadImage = () => {
-        const image = loadImage(this.props.src, this.handleLoad);
-        image.onerror = this.handleError;
-    }
-
     handleLoad = (image) => {
         if (this.mounted) {
```

Comment:
Could we get rid of `this.mounted`? Since the handleLoad and handleError are both called by the DOM now, they would only be called when the component is mounted

Reply:
I just tested this. Nope this would still call the `setState` below even if the component is removed after image load. 
We cannot cancel it without using image constructor or changing the src to empty string which seems like an odd pattern to store src in state and trigger update when unmounting component.